doc/source/howtos/proxysql-v1.rst

```diff
@@ -0,0 +1,424 @@
+
+Use the ``--enable`` option to automatically configure a |PXC| node
+into ProxySQL.
+The ``proxysql-admin`` tool will do the following:
+
+* Add |PXC| node into the ProxySQL database
+
+* Add the ``proxysql_galera_checker`` monitoring script
+  into the ProxySQL ``scheduler`` table if it is not available.
+  This script checks for desynced nodes and temporarily deactivates them.
+  It also calls the ``proxysql_node_monitor`` script,
+  which checks cluster node membership
+  and re-configures ProxySQL if the membership changes.
+
+* Create two new |PXC| users with the ``USAGE`` privilege on the node
+  and add them to ProxySQL configuration, if they are not already configured.
+  One user is for monitoring cluster nodes,
+  and the other one is for communicating with the cluster.
+
+.. note::
+
+   Please make sure to use super user credentials from Cluster
+   to setup the default users.
+
+The following example shows how to add a |PXC| node
+using the ProxySQL configuration file
+with all necessary connection and authentication information:
+
+.. code-block:: bash
+
+   $ proxysql-admin --config-file=/etc/proxysql-admin.cnf --enable
+   
+   This script will assist with configuring ProxySQL (currently only Percona XtraDB cluster in combination with ProxySQL is supported)
+   
+   ProxySQL read/write configuration mode is singlewrite
+   
+   Configuring ProxySQL monitoring user..
+   ProxySQL monitor username as per command line/config-file is monitor
+   
+   User 'monitor'@'127.%' has been added with USAGE privilege
+   
+   Configuring the Percona XtraDB Cluster application user to connect through ProxySQL
+   Percona XtraDB Cluster application username as per command line/config-file is proxysql_user
+   
+   Percona XtraDB Cluster application user 'proxysql_user'@'127.%' has been added with the USAGE privilege, please make sure to the grant appropriate privileges
+   
+   Adding the Percona XtraDB Cluster server nodes to ProxySQL
+   
+   Configuring singlewrite mode with the following nodes designated as priority order:
+   
+   Write node info
+   +-----------+--------------+-------+---------+---------+
+   | hostname  | hostgroup_id | port  | weight  | comment |
+   +-----------+--------------+-------+---------+---------+
+   | 127.0.0.1 | 10           | 25000 | 1000000 | WRITE   |
+   +-----------+--------------+-------+---------+---------+
+   
+   ProxySQL configuration completed!
+   
+   ProxySQL has been successfully configured to use with Percona XtraDB Cluster
+   
+   You can use the following login credentials to connect your application through ProxySQL
+   
+   mysql --user=proxysql_user -p --host=localhost --port=6033 --protocol=tcp
+
+Disabling ProxySQL
+------------------
+
+Use the ``--disable`` option to remove a |PXC| node's configuration
+from ProxySQL.
+The ``proxysql-admin`` tool will do the following:
+
+* Remove |PXC| node from the ProxySQL database
+* Stop the ProxySQL monitoring daemon for this node
```

Comment:
Will also:
Remove the application user for this cluster
Removes any query rules setup for this cluster